### DISCUSSION TITLE
Trame Network Interface

### DISCUSSION THREAD

**Discussion by tmarrinan** [Kitware/trame#612] [General] [answered]
Hello.
I have a Trame app that works when I run it locally. However, I am now trying to run on a remote server and running into issues. I select a port that allows traffic on the server's firewall using the `--port` option, but when the Trame app launches it says:

```
App running at:
 - Local:   http://localhost:8008/
 - Network: http://127.0.0.1:8008/
```

Again, traffic is allowed no problem when running other apps on that same port (e.g. `python3 -m http.server 8008`), but I receive an error about not being able to connect when it is the Trame server.

Notice that the "Network" interface lists the local IP 127.0.0.1 instead of the actual remotely accessible IP address.

If I run this simple Python script on the same server, it properly prints out the remotely accessible IP address:

```
import socket
hostname = socket.gethostname()
print(socket.gethostbyname(hostname))
```

**Comment by jourdain** [comment, marked as answer]
Add `--host 0.0.0.0` to expose the serve to all network interfaces. 
Also for real deployment, you should embed your trame app inside docker to enable multi-user separation. 

If you really want to enable a single trame instance to be your server, you should add a `--timeout 0` to prevent it from shutting down automatically after 300 seconds.
